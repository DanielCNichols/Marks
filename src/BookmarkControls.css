.sort {
  width: 20%;
  padding: 5px;
  border-radius: 6px;
  background-color: rgb(64, 65, 64);
  border: 2px solid rgba(255, 255, 255, .5);
  color: rgba(255, 255, 255, .8);
  font-family: inherit;
}


.sort-select {
  height: 35px;
  border: 2px solid rgba(255, 255, 255, .5);
  font-family: inherit;
  border-radius: 0;
  background-color: rgb(64, 65, 64);
  color: rgba(255, 255, 255, .8)
}


@media screen and (min-width: 450px) {
  .sort {
    width: 75px;
  }
}
