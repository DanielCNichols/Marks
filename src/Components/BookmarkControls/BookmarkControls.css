.sort {
  width: 20%;
  padding: 5px;
  border-radius: 6px;
  background-color: rgb(64, 65, 64);
  border: 2px solid rgba(255, 255, 255, 0.5);
  color: rgba(255, 255, 255, 0.8);
  font-family: inherit;
}

.sort-select {
  width: 100%;
  height: 35px;
  border: 2px solid rgba(255, 255, 255, 0.5);
  font-family: inherit;
  border-radius: 0;
  background-color: rgb(64, 65, 64);
  color: rgba(255, 255, 255, 0.8);
}
