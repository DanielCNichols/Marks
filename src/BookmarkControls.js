import React from 'react';
import './BookmarkControls.css';

export default function BookmarkControls(props) {
  const handleSort = function(ev) {
    console.log(ev.target.sort.value);
    ev.preventDefault();
    props.sort(ev.target.sort.value);
  };

  return (
    <div>
      <form onSubmit={handleSort}>
        <label>
          Sort by rating
          <select
            className="sort-select"
            name="sort"
            aria-label="Sort by rating"
          >
            <option value="asc">Ascending</option>
            <option value="desc">Descending</option>
          </select>
        </label>
        <button className="sort">Sort</button>
      </form>
    </div>
  );
}
