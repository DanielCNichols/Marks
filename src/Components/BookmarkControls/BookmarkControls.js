import React from 'react';
import './BookmarkControls.css';

export default function BookmarkControls({ sort }) {
  const handleSort = val => {
    sort(val);
  };

  return (
    <div>
      <label>
        Sort by rating
        <select
          className="sort-select"
          name="sort"
          aria-label="Sort by rating"
          onChange={ev => handleSort(ev.target.value)}
        >
          <option value="">Sort by</option>
          <option value="asc">Ascending</option>
          <option value="desc">Descending</option>
        </select>
      </label>
    </div>
  );
}
