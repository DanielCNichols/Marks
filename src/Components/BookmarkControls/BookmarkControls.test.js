import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import BookmarkControls from './BookmarkControls';

describe('BookmarkControlsu tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookmarkControls />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer.create(<BookmarkControls />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
