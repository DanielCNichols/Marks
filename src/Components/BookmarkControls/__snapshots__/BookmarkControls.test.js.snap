// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkControlsu tests renders the UI as expected 1`] = `
<div>
  <select
    aria-label="Sort by rating"
    className="sort-select"
    name="sort"
    onChange={[Function]}
  >
    <option
      value=""
    >
      Sort by rating
    </option>
    <option
      value="asc"
    >
      Ascending
    </option>
    <option
      value="desc"
    >
      Descending
    </option>
  </select>
</div>
`;
